examples/compiler/03_iet.ipynb

```diff
@@ -0,0 +1,430 @@
+      "{\n",
+      "  float (*restrict u)[x_size + 1 + 1][y_size + 1 + 1] __attribute__((aligned(64))) = (float (*)[x_size + 1 + 1][y_size + 1 + 1]) u_vec;\n",
+      "  /* Flush denormal numbers to zero in hardware */\n",
+      "  _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);\n",
+      "  _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);\n",
+      "  for (int time = time_m, t0 = (time)%(2), t1 = (time + 1)%(2); time <= time_M; time += 1, t0 = (time)%(2), t1 = (time + 1)%(2))\n",
+      "  {\n",
+      "    struct timeval start_section0, end_section0;\n",
+      "    gettimeofday(&start_section0, NULL);\n",
+      "    for (int x = x_m; x <= x_M; x += 1)\n",
+      "    {\n",
+      "      #pragma omp simd\n",
+      "      for (int y = y_m; y <= y_M; y += 1)\n",
+      "      {\n",
+      "        u[t1][x + 1][y + 1] = u[t0][x + 1][y + 1] + 2;\n",
+      "      }\n",
+      "    }\n",
+      "    gettimeofday(&end_section0, NULL);\n",
+      "    timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;\n",
+      "  }\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(op)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When we take a look at the constructed expression ($u[t1][x + 1][y + 1] = u[t0][x + 1][y + 1] + 2$) we see $+ 1$ being added to the spatial indexes of $u$:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That occurs because we have grid points surrounding the domain region, i.e. `ghost` points that are accessed by the stencil when iterating in the proximity of the domain boundary. This region is called `Halo`. The Halo region can be seen below, it's the zeros surrounding the domain region."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[[0. 0. 0. 0. 0.]\n",
+      "  [0. 1. 1. 1. 0.]\n",
+      "  [0. 1. 1. 1. 0.]\n",
+      "  [0. 1. 1. 1. 0.]\n",
+      "  [0. 0. 0. 0. 0.]]\n",
+      "\n",
+      " [[0. 0. 0. 0. 0.]\n",
+      "  [0. 1. 1. 1. 0.]\n",
+      "  [0. 1. 1. 1. 0.]\n",
+      "  [0. 1. 1. 1. 0.]\n",
+      "  [0. 0. 0. 0. 0.]]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(u.data_with_halo)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Thus, the array accesses become logically aligned to the equation’s natural domain. For instance, given the usual Function $u(t, x)$ having one point on each side of the $x$ halo region, the array accesses $u[t, x]$ and $u[t, x + 2]$ are transformed, respectively, into $u[t, x + 1]$ and $u[t, x + 3]$. When $x = 0$, therefore, the values $u[t, 1]$ and $u[t, 3]$ are fetched, representing the first and third points in the computational domain. \n",
```

Comment:
since the Operator above is over a 2D grid, here you should use a 2D ``u`` too (i.e., ``u(t, x, y)``)